imapclient/client.go

```diff
@@ -962,7 +962,7 @@ func (ce *commandEncoder) flush() {
 // Literal encodes a literal.
 func (ce *commandEncoder) Literal(size int64) io.WriteCloser {
 	var contReq *imapwire.ContinuationRequest
-	if size > 4096 || !ce.client.Caps().Has(imap.CapLiteralMinus) {
+	if size > 4096 || !ce.client.caps.Has(imap.CapLiteralMinus) {
```

Comment:
We need to acquire the client mutex when looking up `client.caps`.

Reply:
I see, therefore calling Caps() may be a better solution.

Reply:
No, calling `Caps` may result in a roundtrip, which is undesirable here.